### PR TITLE
CB-13143: Integrate and replace SplashScreens with Launch Storyboard

### LINKED ISSUE #770: Use storyboards & default split screens support - required from July 2020
# Feature Request

## Motivation Behind Feature

Apple requires adaptive user interface features from April 2020:

"starting April 2020, apps submitted to the App Store must use an Xcode storyboard to provide the app’s launch screen and must have an interface that supports any display size"

https://developer.apple.com/news/?id=01132020b

## Feature Description

* add default storyboard

* add default config for splitscreen

* remove legacy icon & legacy launch image logic, see

### PR DESCRIPTION
**Pull request by dpogue**
### Platforms affected
iOS

### Motivation and Context

1. Apple is deprecating Launch Images and will require Launch Storyboards for all apps as of April 2020 (Closes #770).

2. As per the decisions resulting from the plugin audit, [CB-13143](https://issues.apache.org/jira/browse/CB-13143) intends to integrate the SplashScreen plugin API into each native platform and deprecate the plugin.

### Description

* Remove the default Launch Images
* Remove support for Launch Images from the `prepare` step
* Integrate the required functionality for showing/hiding a screenshot of the Launch Storyboard to `CDVViewController`
* Add a built-in plugin with the same API as the SplashScreen plugin

### Testing

* Manually created a test app and verified the launch storyboard behaviour.
* Manually confirmed the SplashScreen API works as intended from the JS side.

### Checklist

- [ ] I've run the tests to see all new and existing tests pass
- [ ] I added automated test coverage as appropriate for this change
- [x] If this Pull Request resolves an issue, I linked to the issue in the text above (and used the correct [keyword to close issues using keywords](https://help.github.com/articles/closing-issues-using-keywords/))
- [ ] I've updated the documentation if necessary

### CODE REVIEW

**Comment by erisu** [comment]
@dpogue I pushed an update to remove the unit tests that I forgot to remove in the commit that you cherry picked. `refactor (prepare): delete legacy splash screen launch images`. The Jasmine unit tests should be OK now.

Locally I saw another error but from the building test.

**Comment by dpogue** [comment]
@erisu FYI I rebased and squashed your Jasmine test fix commit into the one where the prepare code was removed.

This should hopefully pass all the Obj-C tests too

**Comment by erisu** [comment]
During testing, this PR will also no longer provide or display the default Cordova splashscreen as designed.

**Comment by erisu** [comment]
Documentation would need to go into `cordova-doc`.

Only the plugin documentation were written within its own repos and then the `cordova-doc` repo would grab it. Platform docs were manually managed directly in `cordova-doc`.

**Comment by kanongil** [comment]
This displays an empty screen for me (transparent, unless backgroundcolor is is). The storyboard is loaded and ok, but the view returned from `snapshotViewAfterScreenUpdates:` is blank.

File: CordovaLib/Classes/Public/CDVViewController.m [hunk 1]
```diff
@@ -729,6 +760,43 @@ - (void)onAppDidEnterBackground:(NSNotification*)notification
...
+
+    if ([self.settings cordovaBoolSettingForKey:@"AutoHideSplashScreen" defaultValue:YES]) {
+       [self showLaunchScreen:NO];
+    }
+}
+
+/**
+ Method to be called from the plugin JavaScript to show or hide the launch screen.
+ */
+- (void)showLaunchScreen:(BOOL)visible
```
Comment (kanongil) [inline CordovaLib/Classes/Public/CDVViewController.m:779]: This method delays showing the splash screen by `SplashScreenDelay` ms when `visible = YES`.

**Comment by erisu** [comment]
@kanongil Please test against nightly or master.

The Issue you mentioning sound relatively similar to what has already been reported and fixed in master.

If you have any issues please open a new ticket instead of messaging inside a PR which has already been merged and released.

**Pull request merged**